### PR TITLE
Add `FixedPointOperand` blanket implementation

### PR DESCRIPTION
**Pull request by juangirini**
This PR replaces the explicit `FixedPointOperand` impls by a blanket one so that `Balance` implements it.

### CODE REVIEW

**Comment by juangirini** [comment]
bot fmt

File: frame/assets/src/types.rs [hunk 1]
```diff
@@ -293,8 +293,8 @@ where
 	T: Config<I>,
 	I: 'static,
 	CON: Convert<BalanceOf<F, T>, AssetBalanceOf<T, I>>,
-	BalanceOf<F, T>: FixedPointOperand + Zero,
-	AssetBalanceOf<T, I>: FixedPointOperand + Zero,
+	BalanceOf<F, T>: Zero,
+	AssetBalanceOf<T, I>: Zero,
```
Comment (gavofyork) [inline frame/assets/src/types.rs:297]:
```suggestion
```

File: frame/assets/src/lib.rs [hunk 2]
```diff
@@ -241,7 +241,8 @@ pub mod pallet {
 			+ Copy
 			+ MaybeSerializeDeserialize
 			+ MaxEncodedLen
-			+ TypeInfo;
+			+ TypeInfo
+			+ FixedPointOperand;
```
Comment (gavofyork) [inline frame/assets/src/lib.rs:245]:
```suggestion
			+ TypeInfo;
```

Comment (gavofyork) [reply frame/assets/src/lib.rs:245]: This should not be needed. `FixedPointOperand` has a blanket impl and all bounds required for it are already bounded by `Copy + AtLeast32BitUnsigned`.

**Review by gavofyork — CHANGES_REQUESTED** [review]
Explicitly stating the bound should not be needed.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3264496

**Comment by juangirini** [comment]
I haven't found that blanket impl so I added it as part of this PR. Also, `Copy + AtLeast32BitUnsigned` are missing `CheckedNeg` compared to `FixedPointOperand` so I added that too.

File: frame/nis/src/lib.rs [hunk 3]
```diff
@@ -224,7 +226,8 @@ pub mod pallet {
 			+ Default
 			+ From<u64>
 			+ TypeInfo
-			+ MaxEncodedLen;
+			+ MaxEncodedLen
+			+ CheckedNeg;
```
Comment (athei) [inline frame/nis/src/lib.rs:230]: Why is this necessary? This PR does not remove any bounds in this pallet. Additionally, this type seems to have a lot of repeated bounds over `trait Balance` through which it is already bound in the `type Currency` above.

Comment (juangirini) [reply frame/nis/src/lib.rs:230]: `CurrencyBalance` is just the `Balance` type + `From<u64>` and therefore it needs to implement `Balance`. Todo it through its blanket impl now we need, the newly added, `CheckedNeg`.
Alternatively, to make it more clear, I guess we could just do 
```rust
type CurrencyBalance: Balance + From<u64>;
```

Comment (athei) [reply frame/nis/src/lib.rs:230]: Got it. Yes. I think your suggestion is good. It would bring it more in line with the doc on it.

**Review by athei — APPROVED** [review]

**Review by pgherveou — APPROVED** [review]

File: frame/assets/src/lib.rs [hunk 4]
```diff
@@ -241,7 +243,8 @@ pub mod pallet {
 			+ Copy
 			+ MaybeSerializeDeserialize
 			+ MaxEncodedLen
-			+ TypeInfo;
+			+ TypeInfo
+			+ CheckedNeg;
```
Comment (gavofyork) [inline frame/assets/src/lib.rs:247]: Avoid adding here - just add it into `BaseArithmetic` trait/blanket impl along with all the other `Checked*` bounds.

**Review by gavofyork — CHANGES_REQUESTED** [review]
Getting there. Left a comment.

**Review by gavofyork — APPROVED** [review]
LGTM.

**Comment by juangirini** [comment]
bot merge
